ui/src/pages/transformations/formula.js

```diff
@@ -17,17 +18,49 @@ import {
 } from '../../components';
 
 
-export const createFormulaRow = (parent, index, output, formula, columnId) => {
+function buildSelectColumnType(index) {
+  return `
+  <select id="datatype-${index}">
+  <option value="string" selected>String</option>
+  <option value="integer">Integer</option>
+  <option value="decimal">Decimal</option>
+  <option value="boolean">Boolean</option>
+  <option value="datetime">Datetime</option>
+  </select>
+  `;
+}
+
+function buildSelectColumnPrecision(index) {
+  return `
+  <select id="precision-${index}">
+  <option value="2" selected>2 decimals</option>
+  <option value="3">3 decimals</option>
+  <option value="4">4 decimals</option>
+  <option value="5">5 decimals</option>
+  <option value="6">6 decimals</option>
+  <option value="7">7 decimals</option>
+  <option value="8">8 decimals</option>
+  </select>
+  `;
+}
+
+export const createFormulaRow = (parent, index, output, formula, dataType, precision) => {
   const item = document.createElement('div');
+  const typeSelect = buildSelectColumnType(index);
+  const precisionSelect = buildSelectColumnPrecision(index);
   item.classList.add('list-wrapper');
   item.id = `wrapper-${index}`;
   item.style.width = '100%';
   item.innerHTML = `
-      <input type="text" placeholder="Output column" style="width: 70%;" ${output ? `value="${output}"` : ''} />
+      <div class="output-group">
+      <input id="output-${index}" type="text" placeholder="Output column" ${output ? `value="${output}"` : ''} />
+      ${typeSelect}
+      ${precisionSelect}
       <button id="delete-${index}" class="button delete-button">DELETE</button>
+      </div>
       <div class="input-group">
-          <label class="label" for="${columnId || `formula-${index}`}">Formula:</label>
-          <textarea id="${columnId || `formula-${index}`}" style="width: 100%;">${formula ? `${formula}` : ''}</textarea>
```

Comment:
Id is needed to pass to backend to identify whether formula is edited

Reply:
Yes sure, but I detected that if we change the name, the column is recreated in the backend side.